HunterPie.Integrations/Datasources/MonsterHunterRise/Entity/Player/MHRPlayer.cs

```diff
@@ -431,22 +478,42 @@ private void GetPlayerDebuffAbnormalities()
                 _ => Process.Memory.Read<int>(debuffsPtr + schema.DependsOn)
             };
 
-            bool isConditionValid = schema.CompareOperator switch
+            MHRDebuffStructure abnormality = new();
+
+            switch (schema.Id)
             {
-                AbnormalityCompareType.WithValue => abnormSubId == schema.WithValue,
-                AbnormalityCompareType.WithValueNot => abnormSubId != schema.WithValueNot,
-                _ => false
-            };
+                case "ABN_LEECHED":
+                    abnormality.Timer = _debuffCondition.HasFlag(DebuffConditions.Blooding) ? 1 : 0;
+                    break;
+                case "ABN_WINDMANTLE":
+                    if (_commonCondition.HasFlag(CommonConditions.WindMantle))
+                    {
+                        abnormality = Process.Memory.Read<MHRDebuffStructure>(debuffsPtr + schema.Offset);
+                        abnormality.Timer /= AbnormalityData.TIMER_MULTIPLIER;
+                        abnormality.Timer = schema.MaxTimer - abnormality.Timer;
+                    }
 
-            MHRDebuffStructure abnormality = new();
+                    break;
+                case "ABN_STUN_BUILDUP":
+                    long subPtr = Process.Memory.Read<long>(debuffsPtr + 0x848);
```

Comment:
Same thing as the other review comment. Keep scanners agnostic.

Reply:
Scanner edited.